xwiki-commons-core/xwiki-commons-jakartabridge/xwiki-commons-jakartabridge-common/src/main/java/org/xwiki/jakartabridge/internal/JavaxToJakartaCollection.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.jakartabridge.internal;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.function.Function;
+
+import org.xwiki.jakartabridge.JakartaBridge;
+
+/**
+ * @param <C> the type of {@link Collection}
+ * @param <X> the javax type
+ * @param <K> the jakarta type
+ * @version $Id$
+ * @since 42.0.0
+ */
+public class JavaxToJakartaCollection<C extends Collection<K>, X, K> extends AbstractJavaxToJakartaWrapper<C>
+    implements Collection<X>
+{
+    private static final Object[] EMPTY_OBJECT_ARRAY = new Object[0];
+
+    protected final Function<K, X> javaxProvider;
+
+    protected final Function<X, K> jakartaProvider;
+
+    /**
+     * @param wrapped the wrapped version
+     * @param javaxProvider the javax wrapper provider
+     * @param jakartaProvider the jakarta wrapper provider
+     */
+    public JavaxToJakartaCollection(C wrapped, Function<K, X> javaxProvider, Function<X, K> jakartaProvider)
+    {
+        super(wrapped);
+
+        this.javaxProvider = javaxProvider;
+        this.jakartaProvider = jakartaProvider;
+    }
+
+    @Override
+    public int size()
+    {
+        return getJakarta().size();
+    }
+
+    @Override
+    public boolean isEmpty()
+    {
+        return getJakarta().isEmpty();
+    }
+
+    @Override
+    public boolean contains(Object o)
+    {
+        return getJakarta().contains(this.jakartaProvider.apply((X) o));
+    }
+
+    @Override
+    public Iterator<X> iterator()
+    {
+        return JakartaBridge.toJavax(getJakarta().iterator(), this.javaxProvider, this.jakartaProvider);
+    }
+
+    @Override
+    public Object[] toArray()
+    {
+        return toArray(EMPTY_OBJECT_ARRAY);
+    }
+
+    @Override
+    public <T> T[] toArray(T[] a)
+    {
+        if (a.length == 0) {
```

Comment:
This looks wrong to me. According to my understanding, a new array should be allocated when the passed array is too small. Also, doesn't this mean that `toArray()` always returns an empty array?

Reply:
This code is indeed wrong. Fixing.